Space2Study-UA-1125/frontEnd
# Update dependencies related to Vite and Vitest

Dependencies related to Vite and Vitest were updated to the latest versions.
Also, `@vitest/coverage-c8` is [deprecated](https://www.npmjs.com/package/@vitest/coverage-c8) and doesn't work with the updated versions of the dependencies, so it was replaced with [`@vitest/coverage-v8`](https://www.npmjs.com/package/@vitest/coverage-v8), as recommended in the [Vitest documentation](https://vitest.dev/guide/coverage#coverage-providers).

To test these changes:

- pull `bugfix/#124/update-vite-and-vitest` branch:
```bash
git pull origin bugfix/#124/update-vite-and-vitest
```
- remove your local `node_modules`:
```bash
rm -rf node_modules
```
- install dependencies:
```bash
npm install
```
- run the test:
```bash
npm test WhatCanYouDo
```
- optionally, you can modify a function in `tests/unit/containers/guest-home-page/WhatCanYouDo.spec.js` by adding `console.log()` to be sure the mock works: 
```javascript
vi.mock('~/containers/guest-home-page/google-button/GoogleButton', () => ({
  __esModule: true,
  default: function () {
    console.log('Mock works')
    return <button>Google</button>
  }
}))
```

If everything works as expected, the test will be passed.
